src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1315,14 +1315,17 @@ void Unit::CalculateMeleeDamage(Unit* victim, CalcDamageInfo* damageInfo, Weapon
         {
             damageInfo->HitInfo     |= HITINFO_GLANCING;
             damageInfo->TargetState  = VICTIMSTATE_HIT;
-            int32 leveldif = int32(victim->GetLevel()) - int32(GetLevel());
+            int32 leveldif = int32(victim->GetLevelForTarget(this)) - int32(GetLevel());
+            if (leveldif == 0)
```

Comment:
Please handle somehow the case of negative leveldif. If that cannot happen, add a comment too mentioning "this code should never happen and it's just a failsafe"

Reply:
```suggestion
            if (leveldif < 0)
            {
            	TC_LOG_DEBUG("entities.unit", "Unit::CalculateMeleeDamage: (Player) %s attacked %s. Glancing should never happen against lower level target", GetGUID().ToString().c_str(), victim->GetGUID().ToString().c_str());
            	break;
            }
            if (leveldif == 0)
```

At least with a log we don't silent the underlying problem.
The `break` will prevent any damages modifications.

Is it good for you ?